### PR TITLE
Decorators

### PR DESCRIPTION
**Pull request by dev89s**
1. Create a Nameable Base Class as an interface
2. Implement method `correct_name` in the class
3. Person class inherits from the Nameable class
4. Implement a `correct_name` method that just returns the name attribute
5. Create NameDecorator class as the base name Decorator interface that inherits from Nameable class
6. This class has an instance variable of `@nameable` which is initialized by a nameable object
7. Implement `correct_name` method that returns the output of `@nameable.correct_name`
8. Create CapitalizeDecorator class that inherits from NameDecorator class
9. Implement `correct_name` method that returns the capitalized output of `@nameable.correct_name`
10. Create TrimmerDecorator class that inherits from NameDecorator class
11. Implement `correct_name` method that returns the trimmed output of `@nameable.correct_name`

### CODE REVIEW

File: decorators.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class CapitalizeDecorator < NameDecorator
+  def correct_name
+    @nameable.correct_name.split(/ /).map(&:capitalize).join(' ')
+  end
+end
+
+class TrimmerDecorator < NameDecorator
+  def correct_name
+    @nameable.correct_name.strip
+  end
```
Comment (gemmen29) [inline decorators.rb]: - [x] Kindly change the implementation of the `correct_name` method in the `TrimmerDecorator` to match this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#:~:text=Implement%20a%20method%20correct_name%20that%20makes%20sure%20that%20the%20output%20of%20%40nameable.correct_name%20has%20a%20maximum%20of%2010%20characters.%20If%20it%27s%20longer%20it%20should%20trim%20the%20word.). The output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @dev89s, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by whiz25 — APPROVED** [review]
Hi @dev89s, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- `correct_name` method returns a maximum of 10 characters as suggested

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — COMMENTED** [review]
Hi @dev89s,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* You requested review for set up associations but this is the decorator project. Kindly submit the correct form/project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**